File: implementation/scipamato/scipamato-public-persistence-jooq/src/test/java/ch/difty/scipamato/public_/persistence/codeclass/JooqCodeClassServiceTest.java
package ch.difty.scipamato.public_.persistence.codeclass;

import static org.assertj.core.api.Assertions.*;
import static org.mockito.Mockito.*;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import ch.difty.scipamato.public_.entity.CodeClass;

@RunWith(MockitoJUnitRunner.class)
public class JooqCodeClassServiceTest {

    private final JooqCodeClassService service = new JooqCodeClassService();

    @Mock
    private CodeClassRepository repoMock;

    @Test
    public void findingCodeClasss_delegatesToRepo() {
        service.setRepository(repoMock);

        String languageCodeClass = "de";

        List<CodeClass> ccs = new ArrayList<>();
        ccs.add(new CodeClass(1, "en", "cc1", ""));
        ccs.add(new CodeClass(2, "en", "cc2", ""));
        when(repoMock.find(languageCodeClass)).thenReturn(ccs);

        assertThat(extractProperty(CodeClass.NAME).from(service.find(languageCodeClass))).containsOnly("cc1", "cc2");

        verify(repoMock).find(languageCodeClass);

        verifyNoMoreInteractions(repoMock);
    }

}

File: public/public-persistence-jooq/src/test/kotlin/ch/difty/scipamato/publ/persistence/codeclass/JooqCodeClassServiceTest.kt
package ch.difty.scipamato.publ.persistence.codeclass

import ch.difty.scipamato.publ.entity.CodeClass
import io.mockk.confirmVerified
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import org.amshove.kluent.shouldContainSame
import org.junit.jupiter.api.Test

internal class JooqCodeClassServiceTest {

    private val languageCodeClass = "de"

    private val repoMock = mockk<CodeClassRepository>()
    private val service = JooqCodeClassService(repoMock)

    private val ccs = listOf(
        CodeClass(1, "en", "cc1", ""),
        CodeClass(2, "en", "cc2", "")
    )

    @Test
    fun findingCodeClass_delegatesToRepo() {
        every { repoMock.find(languageCodeClass) } returns ccs
        service.find(languageCodeClass).map { it.name } shouldContainSame listOf("cc1", "cc2")
        verify { repoMock.find(languageCodeClass) }
        confirmVerified(repoMock)
    }
}
